include/seqan3/range/detail/inherited_iterator_base.hpp

```diff
@@ -183,11 +188,17 @@ class inherited_iterator_base : public std::conditional_t<std::is_pointer_v<base
         return *this_derived();
     }
 
+    //!\brief Post-increment of non-copyable iterators returns void.
+    constexpr void operator++(int) noexcept(noexcept(++std::declval<derived_t &>()))
+    {
+        ++(*this_derived());
```

Comment:
Well, this is only true for input iterator right. So since this is a base class to inherit from input iterators might never fulfill the Cpp17IteratorConcept. Not sure, that we want this for the generic base class. So Cpp17Iterator returns a copy of the dereferenced element  before the increment. Have you considered this?

Reply:
The version you are talking about is immediately below it. It is more refined than this overload because it requires Semiregular. Since all ForwardIterators are semirgeular, it will use the other overload. But it will even use the other one if the iterator is input-iterator and semiregular which is what we want!

Reply:
ok